action.yml

```diff
@@ -25,23 +25,43 @@ runs:
   using: "composite"
   steps:
     - id: oidc-auth
+      uses: nick-fields/retry@v2
+      with:
+        timeout_seconds: 30
+        max_attempts: 3
+        shell: bash
+        command: |
+          OIDC_TOKEN=$(curl --fail -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL" | jq -r '.value')
+          AUTH_RESPONSE=$(curl \
+            -X POST \
+            -H "Authorization: Bearer $OIDC_TOKEN" \
+            -d '{
+                  "target_repo": "${{ fromJSON(github.event.inputs.client_payload).payload.full_repo_path }}",
+                  "jit_event_id": "${{ fromJSON(github.event.inputs.client_payload).payload.jit_event_id }}",
+                  "execution_id": "${{ fromJSON(github.event.inputs.client_payload).payload.execution_id }}"
+            }' \
+            -s -w %{http_code} \
+            ${{ fromJSON(github.event.inputs.client_payload).payload.jit_base_api }}/github/auth
+          )
+          response_code=${AUTH_RESPONSE: -3}
+          echo "Got status_code $response_code"
+          echo "has_tokens=0" >> $GITHUB_OUTPUT
+          if [ $response_code = "200" ]; then
+            response_body=${AUTH_RESPONSE:0:${#AUTH_RESPONSE}-3}
+            echo "github_token=$(echo $response_body | jq -r '.github_token')" >> $GITHUB_OUTPUT
+            echo "jit_token=$(echo $response_body | jq -r '.jit_token')" >> $GITHUB_OUTPUT
+            echo "has_tokens=1" >> $GITHUB_OUTPUT
+          elif [ "$response_code" -eq 503 ] || [ "$response_code" -eq 504 ]; then
+            # service unavailable, gateway timeout
+            echo "Response code is retryable - retrying"
+            exit 1
+          fi
+
+    - name: Verify we have tokens
       shell: bash
+      if: steps.oidc-auth.has_tokens == 0
       run: |
-        OIDC_TOKEN=$(curl --fail -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL" | jq -r '.value')
-        AUTH_RESPONSE=$(curl \
-          -X POST \
-          -H "Authorization: Bearer $OIDC_TOKEN" \
-          --fail \
-          -d '{
-                "target_repo": "${{ fromJSON(github.event.inputs.client_payload).payload.full_repo_path }}",
-                "jit_event_id": "${{ fromJSON(github.event.inputs.client_payload).payload.jit_event_id }}",
-                "execution_id": "${{ fromJSON(github.event.inputs.client_payload).payload.execution_id }}"
-          }' \
-          ${{ fromJSON(github.event.inputs.client_payload).payload.jit_base_api }}/github/auth
-        )
-        
-        echo "github_token=$(echo $AUTH_RESPONSE | jq -r '.github_token')" >> $GITHUB_OUTPUT
-        echo "jit_token=$(echo $AUTH_RESPONSE | jq -r '.jit_token')" >> $GITHUB_OUTPUT
+        exit 99
```

Comment:
why 99 and not 1?

Reply:
wanted an error code that we can identify when we see message on the resource management channel

Reply:
wanted an error code that we can identify when we see message on the resource management channel